tests/test_browser.py

```diff
@@ -112,6 +112,31 @@ def test__request_file(httpbin):
     assert "multipart/form-data" in response.request.headers["Content-Type"]
 
 
+def test__request_file_none(httpbin):
+    """Make sure that a blank file input submits an empty file."""
+    form_html = """
+    <form method="post" action="{}/post">
+      <input name="doc" type="file" />
+    </form>
+    """.format(httpbin.url)
+    form = BeautifulSoup(form_html, "lxml").form
+
+    browser = mechanicalsoup.Browser()
+    response = browser._request(form)
```

Comment:
This is unit testing an internal method. I think it would be better to have an end to end test using httpbin.

Reply:
I probably am misunderstanding this, but doesn't this already send a request to httpbin using a form manufactured and then checking the content of the response ?

Reply:
It does. Personally, I think it's fine to leave as a unit test instead of an end-to-end test, especially since it should be converted into a parameterization of the existing `test__request_file` test (@moy can overrule me here if he disagrees).

Reply:
Moving to a parameterized `test__request_file` is a good argument. But as a standalone test, since we're already paying the performance price for a network round-trip (possibly local), the cost/benefit ratio of testing the user-facing logic together with it would be good.